File: Java/242-ValidAnagram.java
public class Solution {
    public boolean isAnagram(String s, String t) {
        if (s == null || t == null || s.length() != t.length()){
            return false;
        }
        char[] chars = s.toCharArray();
        char[] chart = t.toCharArray();
        Arrays.sort(chars);
        Arrays.sort(chart);
        s = new String(chars);
        t = new String(chart);
        return s.equals(t);
    }

    public boolean isAnagramSol2(String s, String t) {
        if (s.length() != t.length()){
            return false;
        }
        int[] count = new int['z' - 'a' + 1];
        for (char c : s.toCharArray()){
            count[c - 'a']++;
        }
        for (char d : t.toCharArray()){
            count[d - 'a']--;
        }
        for (int i = 0; i < count.length; i++){
            if (count[i] != 0){
                return false;
            }
        }
        return true;
    }    
} 

File: cpp/242-ValidAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        unordered_map<char, int> hm;
        for (char c : s) {
            if (hm.find(c) != hm.end()) {
                hm[c]++;
            }else{
                hm[c] = 1;
            }
        }
        for (char c : t) {
            if (hm.find(c) == hm.end() || hm[c] <= 0) {
                return false;
            }else {
                hm[c]--;
            }
        }
        for (auto i : hm) {
            if (i.second) {
                return false;
            }
        }
        return true;
    }
};
